### PR TITLE
Adjust implementation of user-defined conversion resolution according to the latest LDM decision

### LINKED ISSUE #61470: Adjust implementation of user-defined conversion resolution according to the latest LDM decision
https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-04-06.md#resolution-rules-for-conversion-operators
https://github.com/dotnet/roslyn/issues/56753

### PR DESCRIPTION
**Pull request by AlekseyTs**
Closes #61470.
Related to #56753.

Test plan https://github.com/dotnet/roslyn/issues/60968

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@RikkiGibson, @333fred, @dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@RikkiGibson, @333fred, @dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs [hunk 1]
```diff
@@ -29953,13 +29953,16 @@ .locals init (int V_0)
         }
 
         [Theory]
-        [CombinatorialData]
-        public void ConsumeAbstractConversionOperator_11([CombinatorialValues("implicit", "explicit")] string op)
+        [InlineData("implicit", false)]
+        [InlineData("implicit", true)]
+        [InlineData("explicit", false)]
+        [InlineData("explicit", true)]
+        public void ConsumeAbstractConversionOperator_11([CombinatorialValues("implicit", "explicit")] string op, bool needCast)
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs:29960]: Can we remove `CombinatorialValues`?
nit: `useCast` would make more sense than `needCast`

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Comment by AlekseyTs** [comment]
@RikkiGibson, @333fred, @dotnet/roslyn-compiler For the second review.

**Review by 333fred — APPROVED** [review]
LGTM (commit 2)

**Pull request merged**